File: CollatzConjecture.py
from timeit import default_timer as timer

def Main(n,s):
    for i in range(n,s):
        c = 0
        p = n
        while n >1:
            if (n%2==0):
                n = n/2;c +=1
            else:
                n = 3*n+1;c +=1
        print (f"Num: {p} = Steps: {c}");n = p+1 
  
if __name__ == '__main__':
    print(f"Welcome to Collatz Conjecture Calculator \n")
    n=int(input(f"Please enter a starting number: "))
    s=int(input(f"Please enter a ending number: "))+1
    start = timer()
    Main(n,s)
    end = timer();print(end - start,"Seconds")

File: CollatzConjecture.cpp
#include <iostream>
int num; int end; int count; int pnum;

int main() {
	std::cout << "Please enter a starting number: "; std::cin >> num;
	std::cout << "Please enter a ending number: "; std::cin >> end;
	if (num == 1) { num++; count = 0; }
	else { count = 1; }
	for (int i = (end+1)- num; i > 0; i--) {
		pnum = num;
		do {
			if (num % 2 == 0) { num = num / 2; count += 1; }
			else { num = 3 * num + 1; count += 1; }
		} 		while (num > 1);
		std::cout << "Number = " << pnum << " Steps = " << count << "\n";
		num = pnum + 1;
		count = 1;
	}
	return 0;
}
